runtime/doc/lsp.txt

```diff
@@ -24,88 +24,84 @@ QUICKSTART                                              *lsp-quickstart*
 Nvim provides an LSP client, but the servers are provided by third parties.
 Follow these steps to get LSP features:
 
-  1. Install the nvim-lspconfig plugin.  It provides common configuration for
-     various servers so you can get started quickly.
-     https://github.com/neovim/nvim-lspconfig
-  2. Install a language server. A list of language servers can be found here:
+  1. Install language servers using your package manager or by
+     following the upstream installation instruction.
+
+     A list of language servers is available at:
+
      https://microsoft.github.io/language-server-protocol/implementors/servers/
-     See individual server documentation for installation instructions.
-  3. Add `lua require('lspconfig').xx.setup{…}` to your init.vim, where "xx" is
-     the name of the relevant config. See the nvim-lspconfig README for details.
-     NOTE: Make sure to restart nvim after installing and configuring.
-  4. Check that an LSP client has attached to the current buffer:  >
 
-      :lua print(vim.inspect(vim.lsp.buf_get_clients()))
+  2. Configure the LSP client per language server.
+     A minimal example:
+>
+     vim.lsp.start({
+        name = 'my-server-name',
+        cmd = {'name-of-language-server-executable'},
+        root_dir = vim.fs.dirname(vim.fs.find('.git')[1]),
```

Comment:
As `.git` has been the single most problematic root marker for lspconfig (especially for people having a `.git` in their home directory), maybe it would be better to replace this with a more clear-cut one (e.g., `compile-commands.json` or `.stylua.toml`) throughout this documentation?

Reply:
+1 for me on using something other than `.git`

but I'd go for

- `.clangd` instead of `compile_commands.json`, see https://clangd.llvm.org/config#files, _noevim's repo which has `compile_commands.json` under `build/`_
- `.luarc.json` or instead of `.stylua.toml`, see https://github.com/sumneko/lua-language-server/wiki/Setting

other worthy considerations

- `tsconfig.json` or `.eslintrc`
- `pyproject.toml`
- `go.mod`
- `cargo.toml`